pkg/ipam/allocator/podcidr/podcidr_test.go

```diff
@@ -2096,5 +2096,5 @@ func (s *PodCIDRSuite) TestNewNodesPodCIDRManager(c *C) {
 	// Wait for the controller to try more times, the number of deletedCalls
 	// should not be different because we have successfully deleted the node.
 	time.Sleep(2 * time.Second)
-	c.Assert(onDeleteCalls, checker.Equals, 2)
+	c.Assert(onDeleteCalls, Equals, 2)
```

Comment:
The commit states:

> Change existing use of checker.Equals to use simple Equals or HasLen
instead when possible.

I understand changing the checker method on assertions using protobuf `Message` in other tests, but I'm not sure I understand why we're changing the checker method for regular types such as the one pointed out.

Reply:
While going through the uses of `checker.Equals` to figure out which ones need to be changed to `checker.ExportedEquals` I noted that it was also used for trivial types while there is no need to. It's like using `checker.DeepEquals` to check for trivial types like `int`s and `bool`s, which, while works, is an overkill and IMO should not be encouraged. So while these changes are not essential, they resulted from the process I went through in this PR and hopefully result in simpler testing code when these tests are being copied&pasted to build new tests.

Reply:
Ah thanks for clarifying. I was missing the context that `checker.Equals` == `check.DeepEquals`, the latter which comes from our testing framework. I guess we should have named `checker.Equals` `DeepEquals` instead, but alas :shrug:

Reply:
We already have `checker.DeepEquals` :-)
```
// DeepEquals is a GoCheck checker that does a diff between two objects and
// pretty-prints any difference between the two. It can act as a substitute
// for DeepEquals.
```

It actually uses `reflect.DeepEqual()` but adds use of `comparator.CompareWithNames()` when there is a difference.

`checker.Equals` is similar but instead of using `reflect.DeepEqual()` it uses `cmp.Diff()` that short-circuits the member-by-member deep comparison if the type implements `Equal()`. This allows comparing structures that have circular references deeper in the internal structure. `cmp.Diff()` does not allow comparison of unexported fields by default, so there is some logic to enable that, but unfortunately that logic went into infinite loop with circular data structures :-( At first I tried fixing this but found that not to be enough for comparing the new release of protobuf internals. That's why I added the new `checker.ExportedEquals` that only compares exported fields. This works for protobufs as all the significant differences for the users of protobufs are in exported fields. This avoids reaching into the protobuf implementation details that do have circular references.